src/uct/base/uct_iface.c

```diff
@@ -1034,6 +1038,168 @@ int uct_iface_local_is_reachable(uct_iface_local_addr_ns_t *addr_ns,
     return 1;
 }
 
+struct route_info {
+    int if_index;
+    int family;
+
+    union {
+        struct in_addr  ipv4;
+        struct in6_addr ipv6;
+    } remote_addr;
+
+    int prefix_len;
+    int reachable;
+};
+
+static int netlink_parse_rtattr(struct rtattr *attrs[], int max,
+                                struct rtattr *rta, int len)
+{
+    memset(attrs, 0, sizeof(struct rtattr*) * (max + 1));
+
+    for (; RTA_OK(rta, len); rta = RTA_NEXT(rta, len)) {
+        if (rta->rta_type <= max) {
+            attrs[rta->rta_type] = rta;
+        }
+    }
+
+    return 0;
+}
+
+static void create_ipv6_mask(struct in6_addr *mask, unsigned char prefix_len)
+{
+    int i;
+    for (i = 0; i < 16; i++) {
+        if (prefix_len >= 8) {
+            mask->s6_addr[i] = 0xFF;
+            prefix_len      -= 8;
+        } else if (prefix_len > 0) {
+            mask->s6_addr[i] = (0xFF00 >> prefix_len) & 0xFF;
+            prefix_len       = 0;
+        } else {
+            mask->s6_addr[i] = 0;
+        }
+    }
+}
+
+static void parse_nl_route_cb(struct nlmsghdr *nlh, void *arg)
+{
+    struct route_info *info = (struct route_info*)arg;
+    struct rtmsg *rtm       = NLMSG_DATA(nlh);
+    struct rtattr *rta[RTA_MAX + 1];
+    int *oif;
+
+    rtm->rtm_family = info->family;
+    rtm->rtm_table = RT_TABLE_MAIN;
+
+    if (rtm->rtm_family != info->family) {
+        return;
+    }
+
+    netlink_parse_rtattr(rta, RTA_MAX, RTM_RTA(rtm), RTM_PAYLOAD(nlh));
+
```

Comment:
change this function to `ucs_netlink_get_route_info` and return only relevant attributes (iface index + dst address)